lms/templates/dashboard.html

```diff
@@ -167,20 +163,23 @@
                     if not next_session:
                         continue
                     enrollment = CourseEnrollment(user=user, course_id=next_session['key'], mode=next_session['type'])
+                  # Only show email settings for entitlement cards if the entitlement has an associated enrollment
+                  show_email_settings = entitlement_session and (entitlement_session in show_email_settings_for)
 
                 session_id = enrollment.course_id
                 show_courseware_link = (session_id in show_courseware_links_for)
                 cert_status = cert_statuses.get(session_id)
                 can_refund_entitlement = entitlement and entitlement.is_entitlement_refundable()
                 can_unenroll = (not cert_status) or cert_status.get('can_unenroll') if not unfulfilled_entitlement else False
                 credit_status = credit_statuses.get(session_id)
-                show_email_settings = (session_id in show_email_settings_for)
+                if not entitlement:
+                  show_email_settings = (session_id in show_email_settings_for)
```

Comment:
I think it was cleaner before when this was set above line 146 and only modified if entitlement is present. Can you clarify why it was changed?

Reply:
the session_id was not yet defined at the definition of show_email_settings before

Reply:
also session_id can't be defined until after the `if entitlement` block because of the enrollment logic that happens within it